### PR TITLE
Reassign-billing-project takes JSON body now

### PR DESCRIPTION
**Pull request by helgridly**
...instead of query params.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala [hunk 1]
```diff
@@ -40,10 +40,11 @@ trait AdminApiService extends HttpService with PerRequestCreator with UserInfoDi
     } ~
     path("admin" / "project" / "registration") {
       post {
-        parameters("project", "bucket") { (project, bucket) => requestContext =>
+        entity(as[RawlsBillingProjectTransfer]) { xfer =>
+        requestContext =>
           perRequest(requestContext,
             UserService.props(userServiceConstructor, userInfo),
-            UserService.AdminRegisterBillingProject(RawlsBillingProjectName(project), userInfo.userEmail, bucket))
+            UserService.AdminRegisterBillingProject(RawlsBillingProjectName(xfer.project), RawlsUserEmail(xfer.newOwner), xfer.bucket))
```
Comment (helgridly) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala:47]: This is what inspired this change: this endpoint has to be called by an admin, but we might want to reassign the project to a non-admin. (i.e. dumbledore to hermione)

Comment (kyuksel) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala:47]: good call

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala [hunk 2]
```diff
@@ -117,24 +117,6 @@ class AdminApiServiceSpec extends ApiServiceSpec {
       }
   }
```
Comment (kyuksel) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala:119]: negative testing for invalid JSON may be good

**Review by kyuksel — APPROVED** [review]
Looks good. Didn't test it. 👍ing with suggestion for adding negative testing for JSON body.

File: core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala (hunk 2 above)
Comment (helgridly) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala:119]: Spray should handle that.

**Review by MatthewBemis — APPROVED** [review]

**Review by jmthibault79 — APPROVED** [review]
Looks good

**Pull request merged**